src/results-output/results-doc.ts

```diff
@@ -294,8 +295,29 @@ export function append(text: string, onAppended?: OnAppendedCallback): void {
                         onAppended(new vscode.Location(DOC_URI(), insertPosition),
                             new vscode.Location(DOC_URI(), doc.positionAt(Infinity)));
                     }
-                    if (editQueue.length > 0) {
-                        return append.apply(null, editQueue.shift());
+                    
+                    const queueLength = editQueue.length
+                    if (queueLength > 0) {                        
+                        var numOfQueueItemsWithoutCallback: number = 0;
+                        for (var i=0; i<queueLength; i++) {
+                            var [_, onAppendCallback] = editQueue[i];
+                            if (onAppendCallback != null) {
+                                numOfQueueItemsWithoutCallback = i;
+                                break;
+                            }
+                        }
+                        if (numOfQueueItemsWithoutCallback > 0) {
+                            const maxTextsToGroup = 1000;
+                            var textGroup = [];
+                            for (var i=0; i<Math.min(queueLength, maxTextsToGroup); i++) {
+                                var [text] = editQueue.shift();
+                                textGroup.push(text);
+                            }
+                            return append(textGroup.join('\n'), null)
+                        } else {
+                            var [text, onAppendCallback] = editQueue.shift();
+                            return append(text, onAppendCallback)
+                        }
```

Comment:
Nice work on this. Seems to work fine. I'd like to make it a little more functional, though. I added a suggestion below. If you like it feel free to merge it in. Also feel free to make a suggestion or alter it. This requires importing `takeWhile` from lodash.

If you go with this, I think a test for the logic that gets the `testBatch` (the `takeWhile` and the `map`) could use a unit test. I'll leave that to you, unless you'd like me to do it, which I don't mind.

```suggestion
                    if (editQueue.length > 0) {
                        const batchSize = 1000;
                        const textBatch = takeWhile(editQueue, (value, index) => {
                            return index <= batchSize && !value[1];
                        }).map(value => value[0]);
                        if (textBatch.length > 0) {
                            editQueue.splice(0, textBatch.length);
                            return append(textBatch.join('\n'));
                        } else {
                            return append.apply(null, editQueue.shift());
                        }
                    }
```

Reply:
Nice, it's much more concise now. I remember I was grepping for something like `takeWhile`, but didn't find it being used anywhere in the code, so sticked with that imperative code. Looks much better with `takeWhile` :+1: 

I'll try to find some time to add a few tests in the evening or tomorrow.